specs/lfp-appeals-api-spec.json

```diff
@@ -0,0 +1,375 @@
+{
+  "openapi": "3.0.1",
+  "info": {
+    "title": "Appeals API",
+    "description": "LFP Appeals API Documentation",
+    "version": "1.0.0"
+  },
+  "paths": {
+    "/companies/{company-id}/appeals": {
+      "get": {
+        "tags": [
+          "Appeal"
+        ],
+        "summary": "Get appeals by penalty reference",
+        "operationId": "getAppealsByPenaltyReference",
+        "parameters": [
+          {
+            "name": "company-id",
+            "in": "path",
+            "required": true,
+            "schema": {
+              "type": "string"
+            }
+          },
+          {
+            "name": "penaltyReference",
+            "in": "query",
+            "required": true,
+            "schema": {
+              "type": "string"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Appeals resource retrieved successfully",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "array",
+                  "items": {
+                    "$ref": "#/components/schemas/Appeal"
+                  }
+                }
+              }
+            }
+          },
+          "404": {
+            "description": "Appeals not found",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "array",
+                  "items": {
+                    "$ref": "#/components/schemas/Appeal"
+                  }
+                }
+              }
+            }
+          },
+          "500": {
+            "description": "Internal server error",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "array",
+                  "items": {
+                    "$ref": "#/components/schemas/Appeal"
+                  }
+                }
+              }
+            }
+          }
+        }
+      },
+      "post": {
+        "tags": [
+          "Appeal"
+        ],
+        "summary": "Create a new appeal",
+        "operationId": "submitAppeal",
+        "parameters": [
+          {
+            "name": "ERIC-identity",
+            "in": "header",
+            "required": true,
+            "schema": {
+              "type": "string"
+            }
+          },
+          {
+            "name": "company-id",
+            "in": "path",
+            "required": true,
+            "schema": {
+              "type": "string"
+            }
+          }
+        ],
+        "requestBody": {
+          "content": {
+            "application/json": {
+              "schema": {
+                "$ref": "#/components/schemas/Appeal"
+              }
+            }
+          },
+          "required": true
+        },
+        "responses": {
```

Comment:
Can this be simplified? Why are we returning a location header and an ID in the body?
```
"responses": {
    "201": {
        "description": "Appeal resource created",
        "headers": {
             "location": {
                 "description": "URL to get the Appeal"
             }
        },
        "content": {
                 "description": "ID of the Appeal",
                 "type": "string"
        }
    },
    400: {
           "description": "Invalid appeal"
    },
    401: {
          "User is not authorised to submit this appeal"
    },
    500: {
          "Internal server error"
    }
}
```